src/DynamoDbModel.php

```diff
@@ -195,10 +218,10 @@ public static function all($columns = [], $limit = -1)
         return $model->getAll($columns, $limit);
     }
 
-    public static function first($columns = [])
+    public function first($columns = [])
```

Comment:
One way might be to add a test to getInstance to confirm the calling class matches that of static::$instance. I'm not sure this covers all use cases but it would work in the simplest cases, e.g.

if (is_null(static::$instance) || (get_called_class() != get_class(static::$instance))) {
    static::$instance = new static;
}